modules/generator/generator_test.go

```diff
@@ -129,3 +137,140 @@ func (l testLogger) Log(keyvals ...interface{}) error {
 	l.t.Log(keyvals...)
 	return nil
 }
+
+func BenchmarkPushSpans(b *testing.B) {
+	var (
+		tenant = "test-tenant"
+		reg    = prometheus.NewRegistry()
+		ctx    = context.Background()
+		log    = log.NewNopLogger()
+		cfg    = &Config{}
+
+		walcfg = &storage.Config{
+			Path: b.TempDir(),
+		}
+
+		o = &mockOverrides{
+			processors: map[string]struct{}{
+				"span-metrics":   {},
+				"service-graphs": {},
+			},
+			spanMetricsEnableTargetInfo:             true,
+			spanMetricsTargetInfoExcludedDimensions: []string{"excluded}"},
+		}
+	)
+
+	cfg.RegisterFlagsAndApplyDefaults("", &flag.FlagSet{})
+
+	wal, err := storage.New(walcfg, o, tenant, reg, log)
+	require.NoError(b, err)
+
+	inst, err := newInstance(cfg, tenant, o, wal, reg, log, nil, nil)
+	require.NoError(b, err)
+	defer inst.shutdown()
+
+	req := &tempopb.PushSpansRequest{
+		Batches: []*trace_v1.ResourceSpans{
+			test.MakeBatch(100, nil),
+			test.MakeBatch(100, nil),
+			test.MakeBatch(100, nil),
+			test.MakeBatch(100, nil),
+		},
+	}
+
+	// Add more resource attributes to get closer to real data
+	// Add integer to increase cardinality.
+	// Currently this is about 80 active series
+	// TODO - Get more series
+	for i, b := range req.Batches {
+		b.Resource.Attributes = append(b.Resource.Attributes, []*common_v1.KeyValue{
+			{Key: "k8s.cluster.name", Value: &common_v1.AnyValue{Value: &common_v1.AnyValue_StringValue{StringValue: "test" + strconv.Itoa(i)}}},
+			{Key: "k8s.namespace.name", Value: &common_v1.AnyValue{Value: &common_v1.AnyValue_StringValue{StringValue: "test" + strconv.Itoa(i)}}},
+			{Key: "k8s.node.name", Value: &common_v1.AnyValue{Value: &common_v1.AnyValue_StringValue{StringValue: "test" + strconv.Itoa(i)}}},
+			{Key: "k8s.pod.ip", Value: &common_v1.AnyValue{Value: &common_v1.AnyValue_StringValue{StringValue: "test" + strconv.Itoa(i)}}},
+			{Key: "k8s.pod.name", Value: &common_v1.AnyValue{Value: &common_v1.AnyValue_StringValue{StringValue: "test" + strconv.Itoa(i)}}},
+			{Key: "excluded", Value: &common_v1.AnyValue{Value: &common_v1.AnyValue_StringValue{StringValue: "test" + strconv.Itoa(i)}}},
+		}...)
+	}
+
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		inst.pushSpans(ctx, req)
+	}
+
+	b.StopTimer()
+	runtime.GC()
```

Comment:
Why force a GC after the benchmark has been timed?  Is this to avoid impact on later benchmarks, or to get accurate memory summary below?

Reply:
In this case I was trying to see if the benchmarks could help measure inuse memory, so it is recording `HeapInUse` at the bottom.  Without the GC it just keeps growing in between runs of the benchmark (`-count=5` for example).  So this was an attempt to make the inuse metric more useful.